packages/inferno-animation/src/utils.ts

```diff
@@ -0,0 +1,206 @@
+  if (className) {
+    const tmp = className.split(' ')
+    for (let i=0; i < tmp.length; i++) {
+      node.classList.remove(tmp[i])
+    }
+  }
+}
+
+export function forceReflow () {
+  return document.body.clientHeight;
+}
+
+export function setDisplay(node: HTMLElement, value?: string) {
+  var oldVal = node.style.getPropertyValue('display')
+  if (oldVal !== value) {
+    if (value !== undefined) {
+      node.style.setProperty('display', value)
+    }
+    else {
+      node.style.removeProperty('display')
+      _cleanStyle(node);
+    }
+  }
+  return oldVal
+}
+
+function _cleanStyle(node: HTMLElement) {
+  if (!node.style) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute
+    node.removeAttribute('style')
+  }
+}
+
+export function getDimensions(node: HTMLElement) {
+  var tmpDisplay = node.style.getPropertyValue('display')
+  var isDisplayNone = window.getComputedStyle(node).getPropertyValue('display') === 'none'
+  if (isDisplayNone) {
+    node.style.setProperty('display', 'block')
+  }
+
+  var tmp = node.getBoundingClientRect()
+
+  if (isDisplayNone) {
+    // node.style.display = tmpDisplay
+    node.style.setProperty('display', tmpDisplay)
+    _cleanStyle(node)
+  }
+
+  return {
+    width: tmp.width,
+    height: tmp.height
+  }
+}
+
+export function setDimensions(node: HTMLElement, width: number, height: number) {
+  node.style.width = width + 'px'
+  node.style.height = height + 'px'
+}
+
+export function clearDimensions(node: HTMLElement) {
+  node.style.width = node.style.height = ''
+}
+
+function _getMaxTransitionDuration (/* add nodes as args*/) {
+  let nrofTransitions = 0
+  let maxDuration = 0
+  for (let i=0; i < arguments.length; i++) {
+    const node = arguments[i]
+    if (!node) continue
+
+    const cs = window.getComputedStyle(node)
+    const dur = cs.getPropertyValue('transition-duration').split(',')
+    const del = cs.getPropertyValue('transition-delay').split(',')
+    const props = cs.getPropertyValue('transition-property').split(',').map((prop) => prop.trim())
+    props.forEach((prop) => {
+      if (prop[0] === '-') {
+        let tmp = prop.split('-').splice(2).join('-')
+        if (props.indexOf(tmp) >= 0) {
+          nrofTransitions--
+        }
+      }
+    })
+    const animTimeout = dur.map((v, index) => parseFloat(v) + parseFloat(del[index])).reduce((prev, curr) => prev > curr ? prev : curr, 0)
+
+    nrofTransitions += dur.length
+    if (animTimeout > maxDuration) {
+      maxDuration = animTimeout
+    }
+  }
+
+  return {
+    nrofTransitions: nrofTransitions,
+    maxDuration: maxDuration
+  }
+}
+
+function whichTransitionEvent(){
+  var el = document.createElement('fakeelement');
+  var transitions = {
+    'transition':'transitionend',
+    'OTransition':'oTransitionEnd',
+    'MozTransition':'transitionend',
+    'WebkitTransition':'webkitTransitionEnd'
+  }
+
+  for(let t in transitions){
```

Comment:
this probably does not need any loop at all, and because "transitions" is a constant object we could just do if - return here?

Reply:
actually, it seems the whole function is constant and there is no need for the function at all, or just make it call itself and put the result into a variable.

Reply:
var transitionEndName = (function whichTransitionEvent() { ... })()

Reply:
Will investigate

Reply:
FIXED in 695fe367e